building/build-debs/homeworld-admin-tools/src/setup.py

```diff
@@ -129,7 +129,6 @@ def setup_keygateway(ops: Operations, config: configuration.Config) -> None:
         # keytab is stored encrypted in the configuration folder
         keytab = os.path.join(configuration.get_project(), "keytab.%s.crypt" % node.hostname)
         decrypted = keycrypt.gpg_decrypt_to_memory(keytab)
-        ops.ssh("confirm no existing keytab on @HOST", node, "test", "!", "-e", KEYTAB_PATH)
```

Comment:
The original point here was to avoid overwriting a newer keytab with an older keytab.

We have a few options:

 * Your current change, which allows for the possibility of overwriting and losing a keytab.
 * Silently refuse to overwrite the keytab, which will cause pains when the keytab is updated, but will allow for re-deploying the same cluster.
 * Refuse to overwrite the keytab only if it has changed, which will cause pain (but not as much) when the keytab is updated, and allow for re-deploying the same cluster, but will be more complicated.
 * Refuse to overwrite the keytab only if it has changed and the updated keytab would have an older kvno than the last keytab, which will be the best solution, but will be much more complicated.

I would recommend the third solution, but I'm interested in any thoughts you have.

Reply:
Would it be better to merge the keytabs? From what I understand, you would mainly be concerned about overwriting the existing keytab if there already exists a different keytab for other reasons. Maybe Spire can make a backup of the existing keytab and spit out a warning at the end of the session.

Or is the concern that the administrator may accidentally run the command with an outdated keytab? It doesn't sound like that would be a problem if the latest keytab is stored in, for example, Git.

Reply:
> ...the administrator may accidentally run the command with an outdated keytab? It doesn't sound like that would be a problem...

That sounds like precisely the sort of thing people will do accidentally.

Reply:
I meant that if he does do that, he can just get the correct keytab from Git and replace it.

Reply:
Only if they realize it before their certificates expire. Otherwise, no one can regain access to the cluster until someone goes and drags the disaster recovery keys out of cold storage.

Reply:
Right, it's encrypted with the disaster recovery key. Would merging the keytabs work here, then? It should be relatively simple to do it with ktutil; it's possible to scrape the kvnos from klist, but that feels ugly.

Reply:
Being encrypted with the disaster recovery key has nothing to do with it -- you'd need to pull the DR key out of storage simply because you have no other way to access the cluster.

Merging the keytabs might work, but I don't want that to silently occur. I think your idea of the third solution plus a force-overwrite option sounds good to me.